### PR TITLE
Add filePath to IIterableSource Initialization result

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
On macOS the "proxy icon" updates to show the currently open local file.

**Description**
The desktop app uses filePath to show the "proxy icon" when working with a local file. This change updates PlayerManager to set the representedFilename when a user selects a single file.

### CODE REVIEW

**Comment by defunctzombie** [comment]
@jtbandes I wasn't sure exactly at what level to put filePath support - at first I was thinking it _could_ be an option to IterablePlayer but eventually landed on IIterableSource#initialize since that keeps the sources self-contained and lets you pass a source to the player without it having to know more about it.

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/components/DocumentTitleAdapter.tsx [hunk 1]
```diff
@@ -8,18 +8,14 @@ import {
...
 
 const selectPlayerName = (ctx: MessagePipelineContext) => ctx.playerState.name;
-const selectPlayerFilePath = (ctx: MessagePipelineContext) => ctx.playerState.filePath;
 
 /**
  * DocumentTitleAdapter sets the document title based on the currently selected player
  */
 export default function DocumentTitleAdapter(): JSX.Element {
-  const nativeWindow = useNativeWindow();
   const playerName = useMessagePipeline(selectPlayerName);
```
Comment (jtbandes) [inline packages/studio-base/src/components/DocumentTitleAdapter.tsx:18]: Do you want to move this too? It is very similar in purpose to the representedFilename thing.

**Comment by defunctzombie** [comment]
https://github.com/electron/electron/issues/33647 😭

File: packages/studio-base/src/components/DocumentTitleAdapter.tsx (hunk 1 above)
Comment (defunctzombie) [reply packages/studio-base/src/components/DocumentTitleAdapter.tsx:18]: Maybe another day ♾️

**Pull request merged**